cabal-install/Distribution/Client/CmdInstall.hs

```diff
@@ -130,27 +142,109 @@ installAction (configFlags, configExFlags, installFlags, haddockFlags)
     die' verbosity $ "--enable-benchmarks was specified, but benchmarks can't "
                   ++ "be enabled in a remote package"
 
-  -- We need a place to put a temporary dist directory
+  let verbosity' = lessVerbose verbosity
+  
+  -- First, we need to learn about what's available to be installed.
+  localBaseCtx <- establishProjectBaseContext verbosity' cliConfig
```

Comment:
Does this create the `dist-newstyle` folder in `$PWD`? We should clean up after this in case we are not in a project context if it does.

Reply:
Oh, no. That doesn't create a fake context, that's creating a real context in the current project... but that breaks out of project installs. *Oops.*

Reply:
yeah, I know... I tried real hard to refactor that stuff once. But there is a chicken/egg problem with establishing the project context and file monitor caching in dist-newstyle...